### PR TITLE
feat: katib-db-manager rock

### PR DESCRIPTION
**Pull request by i-chvets**
Katib ROCKs are part of main epic for building secure images using ROCKs. katib-db-manager ROCK is managed by katib-controller charm. Charm's integration tests should be re-used to test functionality of this ROCK.

Summary of changes:
- Initial version of the ROCK.
- Created rockcraft file for katib-db-manager ROCK. ROCK is built and limited manual testing is done.

Manual test:
```
docker run katib-db-manager:v0.15.0_22.04_1 exec /app/katib-db-manager
2023-07-18T18:13:48.097Z [pebble] Started daemon.
2023-07-18T18:13:48.197Z [pebble] POST /v1/exec 98.854491ms 202
2023-07-18T18:13:48.198Z [pebble] GET /v1/tasks/1/websocket/control 152.171µs 200
2023-07-18T18:13:48.199Z [pebble] GET /v1/tasks/1/websocket/stdio 104.532µs 200
2023-07-18T18:13:48.199Z [pebble] GET /v1/tasks/1/websocket/stderr 75.966µs 200
F0718 18:13:48.205851      16 main.go:100] DB_NAME env is not set. Exiting
2023-07-18T18:13:48.315Z [pebble] GET /v1/changes/1/wait 115.753682ms 200
```

### CODE REVIEW

File: katib-db-manager/rockcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      mkdir -p ${CRAFT_PART_INSTALL}/app
+      cd ${CRAFT_PART_SRC}/
+      go mod download all
+      go build -a -o katib-db-manager ./cmd/db-manager/v1beta1
+      install -D -m755 katib-db-manager ${CRAFT_PART_INSTALL}/app/katib-db-manager
+
+      mkdir -p ${CRAFT_PART_INSTALL}/bin
+      curl -o ${CRAFT_PART_INSTALL}/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-${CRAFT_TARGET_ARCH}
+      chmod +x ${CRAFT_PART_INSTALL}/bin/grpc_health_probe
+
```
Comment (i-chvets) [inline katib-db-manager/rockcraft.yaml:47]: Refactor the above health check into its own part.

Comment (i-chvets) [reply katib-db-manager/rockcraft.yaml:52]: Done.

File: katib-db-manager/rockcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+license: Apache-2.0
+build-base: ubuntu:22.04
+base: bare
+run-user: _daemon_
+services:
+  katib-db-manager:
+    override: replace
+    summary: "katib-db-manager service"
+    startup: enabled
+    command: "/app/katib-db-manager"
```
Comment (SamirPS) [inline katib-db-manager/rockcraft.yaml:16]: I think using a working-dir in each of your ROCKs is a good practice if possible

**Review by SamirPS — CHANGES_REQUESTED** [review]

**Comment by i-chvets** [comment]
@SamirPS updated. Please review.

**Review by SamirPS — APPROVED** [review]

**Review by ca-scribner — APPROVED** [review]

**Pull request merged**